src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static int MAX_START_LENGTH = 10;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[MAX_START_LENGTH];
+    }
+
+    private void grow() {
+        T[] newArray = (T[]) new Object[elements.length + elements.length / 2];
+        System.arraycopy(elements, 0, newArray, 0, size);
+        elements = newArray;
+    }
+
+    private void exception(int index) {
+        if (index < 0 || index >= size()) {
+            throw new ArrayListIndexOutOfBoundsException("Index  " + index + " is not "
+                    + "correct for size" + size);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (elements.length == size) {
+            grow();
```

Comment:
if we move `  if (elements.length == size) {` check to  `grow()` method it will reduce code duplication